onnxruntime/core/providers/dml/DmlExecutionProvider/src/GraphDescBuilder.cpp

```diff
@@ -147,8 +147,12 @@ namespace Dml::GraphDescBuilder
                 if (pos != 0)
                 {
                     intermediateEdge.Name = intermediateEdge.Name.substr(0, pos);
+                    intermediateEdge.Name += "nodeIdx:" + std::to_string(intermediateEdge.FromNodeIndex) + "-outputIdx:" + std::to_string(intermediateEdge.FromNodeOutputIndex);
+                }
+                else
+                {
+                    intermediateEdge.Name = "nodeIdx:" + std::to_string(intermediateEdge.FromNodeIndex) + "-outputIdx:" + std::to_string(intermediateEdge.FromNodeOutputIndex);
                 }
-                intermediateEdge.Name += "nodeIdx:" + std::to_string(intermediateEdge.FromNodeIndex) + "-outputIdx:" + std::to_string(intermediateEdge.FromNodeOutputIndex);
```

Comment:
Why duplicate identical code on both lines?

```suggestion
                std::string name = "nodeIdx:" + std::to_string(intermediateEdge.FromNodeIndex) + "-outputIdx:" + std::to_string(intermediateEdge.FromNodeOutputIndex);
                if (pos != 0)
                {
                    intermediateEdge.Name = intermediateEdge.Name.substr(0, pos);
                    intermediateEdge.Name += name;
                }
                else
                {
                    intermediateEdge.Name = name;
                }
```